### PR TITLE
Migrate AnnotationEditor to TS

### PR DESCRIPTION
**Pull request by acelaya**
This PR migrates yet another component from JS to TS.

Does not impact how the component looks or behaves.

### CODE REVIEW

**Review by lyzadanger — COMMENTED** [review]
Looking good! The comments I left are specific to our own conventions. Take a peek and then we can get this landed soon.

File: src/sidebar/components/Annotation/AnnotationEditor.tsx [hunk 1]
```diff
@@ -15,26 +15,24 @@ import TagEditor from '../TagEditor';
...
+import type { TagsService } from '../../services/tags';
+
+type AnnotationEditorProps = {
+  annotation: Annotation;
+  draft: Draft;
+  annotationsService: AnnotationsService;
+  settings: SidebarSettings;
+  toastMessenger: ToastMessengerService;
+  tags: TagsService;
+};
```
[the remark this answers is not in the document]
Comment (acelaya) [reply src/sidebar/components/Annotation/AnnotationEditor.tsx:32]: I completely missed those pre-existing comments. My bad.

**Review by lyzadanger — APPROVED** [review]
Nice work! I look forward to working with you more to get all our stuff converted over.

File: src/sidebar/components/Annotation/AnnotationEditor.tsx [hunk 2]
```diff
@@ -93,10 +89,9 @@ function AnnotationEditor({
     /**
      * Remove tag from draft if present.
      *
-     * @param {string} tag
-     * @return {boolean} Tag removed from draft
+     * @return Tag removed from draft
```
Comment (lyzadanger) [inline src/sidebar/components/Annotation/AnnotationEditor.tsx:92]: Huh. This is interesting. You've kicked up an interesting case here! The removal of the redundant type information makes this comment ambiguous. It sounds like the removed  _tag_ (string) itself is returned, when in fact the return is boolean (FWIW the original `@return` comment was a little vague).

One possibility:

```suggestion
     * @return `true` if tag removed from draft, `false` if tag not found in draft tags
```

Other options:

* Remove this comment entirely as the return logic is pretty easy to scan in this very brief callback function.
* Re-instate the (redundant) type annotation. Hey, we have conventions here, but we like to adapt if they don't apply well!

File: src/sidebar/components/Annotation/AnnotationEditor.tsx [hunk 3]
```diff
@@ -71,11 +69,9 @@ function AnnotationEditor({
     /**
      * Verify `newTag` has content and is not a duplicate; add the tag
      *
-     * @param {string} newTag
-     * @return {boolean} Tag was added to the draft's tags; `false` if duplicate
-     *   or empty
+     * @return Tag was added to the draft's tags; `false` if duplicate or empty
```
Comment (lyzadanger) [inline src/sidebar/components/Annotation/AnnotationEditor.tsx:72]:
```suggestion
     * @return `true` if tag was added to the draft; `false` if duplicate or empty
```

**Review by lyzadanger — CHANGES_REQUESTED** [review]
I regret the thrash on these `@return` comments! See comments. We're close.

File: src/sidebar/components/Annotation/AnnotationEditor.tsx (hunk 2 above)
Comment (acelaya) [reply src/sidebar/components/Annotation/AnnotationEditor.tsx:92]: My personal take on this, I would just remove the comment.

Comments usually have the opposite effect when the code is clear enough to be understood without them.

**Review by lyzadanger — APPROVED** [review]
Yay! Feel free to remove the `@return` comments if you feel they don't add any use.

**Pull request merged**